src/api/composition/CompositionProvider.js

```diff
@@ -19,102 +19,95 @@
  * this source code distribution or the Licensing information page available
  * at runtime from the About dialog for additional information.
  *****************************************************************************/
+import _ from 'lodash';
+import objectUtils from "../objects/object-utils";
 
-define([
-    'lodash',
-    'objectUtils'
-], function (
-    _,
-    objectUtils
-) {
+/**
+ * @typedef {import('../objects/ObjectAPI').DomainObject} DomainObject
+ */
+
+/**
+ * @typedef {import('../objects/ObjectAPI').Identifier} Identifier
+ */
+
+/**
+ * @typedef {import('./CompositionAPI').default} CompositionAPI
+ */
+
+/**
+ * @typedef {import('../../../openmct').OpenMCT} OpenMCT
+ */
+
+/**
+ * A CompositionProvider provides the underlying implementation of
+ * composition-related behavior for certain types of domain object.
+ *
+ * By default, a composition provider will not support composition
+ * modification.  You can add support for mutation of composition by
+ * defining `add` and/or `remove` methods.
+ *
+ * If the composition of an object can change over time-- perhaps via
+ * server updates or mutation via the add/remove methods, then one must
+ * trigger events as necessary.
+ *
+ */
+export default class CompositionProvider {
     /**
-     * A CompositionProvider provides the underlying implementation of
-     * composition-related behavior for certain types of domain object.
-     *
-     * By default, a composition provider will not support composition
-     * modification.  You can add support for mutation of composition by
-     * defining `add` and/or `remove` methods.
-     *
-     * If the composition of an object can change over time-- perhaps via
-     * server updates or mutation via the add/remove methods, then one must
-     * trigger events as necessary.
-     *
-     * @interface CompositionProvider
-     * @memberof module:openmct
+     * @param {OpenMCT} publicAPI
+     * @param {CompositionAPI} compositionAPI
      */
-
-    function DefaultCompositionProvider(publicAPI, compositionAPI) {
+    constructor(publicAPI, compositionAPI) {
+        /** @type {OpenMCT} */
```

Comment:
Ideally these are private, if that is not practical (eg. they are referenced in callbacks or something) at least don't document them. I don't want them exposed in public API.

Reply:
We should be able to make these private. In that case, the JSDoc --> *.d.ts process should pick up on that and mark them as private. The types are here to support type inference within the rest of the class. I'll verify this first-- otherwise we can just remove the docs entirely.

Reply:
Confirming that any methods or params made `#private` are excluded from the declaration files entirely